src/encoded/static/components/item-pages/QualityMetricView.js

```diff
@@ -74,32 +44,49 @@ class QualityMetricViewOverview extends React.PureComponent {
         };
     }
 
+    static getSchemaQCFieldsAsOrderedPairs = memoize(function (schemaProp, context, schemas) {
+
+        if (!schemaProp && !context && !schemas) { return []; }
+
+        let propObj = null;
+        if (schemaProp) {
+            if (schemaProp.type === 'object' && schemaProp.properties) {
+                propObj = schemaProp.properties;
+            } else if (schemaProp.type === 'array' && schemaProp.items && schemaProp.items.type === 'object' && schemaProp.items.properties) {
+                propObj = schemaProp.items.properties;
+            }
+        }
+        
+        if (!propObj && context && schemas) {
+            const typeSchema = schemaTransforms.getSchemaForItemType(
+                schemaTransforms.getItemType(context), schemas || null
+            );
+            if (typeSchema && typeSchema.properties) {
+                propObj = typeSchema.properties;
+            }
+        }
+
+        if (propObj) {
+            return _.chain(propObj)
+                .pick(function (value, key) { return typeof value.qc_order === 'number'; })
+                .pairs()
+                .sortBy(function (pair) { return pair[1].qc_order; })
+                .value();
+        }
+
+        return [];
+    });
+
     render() {
         const { schemas, context } = this.props;
-        const { qcSummary } = this.state;
-
-        //QC metrics
         const tips = object.tipsFromSchema(schemas, context);
-        const commonProps = { 'metric': context, 'tips': tips };
-        const keys = _.difference(_.keys(context), excludedKeys);
-        //keys.push(keys.splice(keys.indexOf('url'), 1)[0]);//move url to end
-        const metrics = keys.length > 0 ?
-            (
-                <div className="overview-list-elements-container">
-                    {_.map(keys, (propKey) => <QCMetricFromEmbed {...commonProps} qcProperty={propKey} />)}
-                </div>
-            ) : (
-                <em>Not Available</em>
-            );
-        //QC metrics summary
-        const metricsSummary = !qcSummary ?
+
+        //QC Metrics Summary
+        const { quality_metric_summary } = context;
+        const qcMetricsSummary = context.quality_metric_summary ?
```

Comment:
I don't like this naming, there is a clear difference yet it is not obvious. Also, why mix snake_case with camelCase?

Reply:
camelCase is convention for JS and snake_case is convention for python; usually snake_case when appears in JS code is a property from JSON response (`context`), which has the snake_case from Python backend & schemas.